bagua/torch_api/contrib/cache_dataset.py

```diff
@@ -0,0 +1,62 @@
+from torch.utils.data.dataset import Dataset
+from .cache_loader import CacheLoader
+
+__all__ = ["CacheDataset"]
+
+
+class CacheDataset(Dataset):
+    def __init__(
+        self,
+        dataset: Dataset,
+        backend: str = "redis",
+        key_prefix: str = "",
+        batch_writes: int = 20,
+        **kwargs,
+    ):
+        """
+        A dataset wrapper which caches `dataset` samples.
```

Comment:
`CachedDataset` wraps a PyTorch `Dataset` to cache its samples in memory, so that accessing these samples after the first time can be much faster. This is useful when samples need tedious preprocessing to produce, or reading the dataset itself is slow, which could slow down the whole training process.